contracts/Core/BondManager.sol

```diff
@@ -0,0 +1,229 @@
+    ) external initialized checkEpochAndState(State.Confirm, epoch, buffer) whenNotPaused {
+        // minJobs=>governance param
+        require(mulJobs.length >= minJobs, "invalid bond creation");
+        require(mulJobs.length <= maxJobs, "number of jobs exceed maxJobs");
+        require(minBond <= bond, "minBond not satisfied");
+
+        numDataBond = numDataBond + 1;
+
+        {
+            uint256 minOccurrence = (mulJobs.length * depositPerJob) / bond;
+            if (minOccurrence == 0) minOccurrence = 1;
+            require(minOccurrence <= occurrence, "not enough bond paid per job");
+        }
+        // slither-disable-next-line reentrancy-benign
+        uint16[] memory jobIds = collectionManager.createMulJob(mulJobs);
+        // slither-disable-next-line reentrancy-benign
+        uint16 collectionId = collectionManager.createCollection(
+            collectionTolerance,
+            collectionPower,
+            uint32(occurrence),
+            collectionAggregation,
+            jobIds,
+            collectionName
+        );
+
+        databonds[numDataBond] = Structs.DataBond(true, collectionId, numDataBond, epoch, msg.sender, jobIds, bond);
+        databondCollections.push(collectionId);
+
+        require(razor.transferFrom(msg.sender, address(this), bond), "invalid transfer");
+    }
+
+    function updateDataBondJob(
+        uint32 bondId,
+        uint8 jobIndex,
+        uint8 weight,
+        int8 power,
+        JobSelectorType selectorType,
+        string calldata selector,
+        string calldata url
+    ) external databondCreatorCheck(bondId, msg.sender) {
+        uint32 epoch = getEpoch();
+        // slither-disable-next-line timestamp
+        require(databonds[bondId].epochBondLastUpdated + epochLimitForUpdateBond <= epoch, "invalid databond update");
+
+        databonds[bondId].epochBondLastUpdated = epoch;
+
+        collectionManager.updateJob(databonds[bondId].jobIds[jobIndex], weight, power, selectorType, selector, url);
+    }
+
+    function updateDataBondCollection(
+        uint32 bondId,
+        uint16 collectionId,
+        uint32 tolerance,
+        uint32 aggregationMethod,
+        int8 power,
+        uint16[] memory jobIds
+    ) external databondCreatorCheck(bondId, msg.sender) {
+        uint32 epoch = getEpoch();
+        require(jobIds.length >= minJobs, "invalid bond updation");
+        require(jobIds.length <= maxJobs, "number of jobs exceed maxJobs");
+        require(databonds[bondId].collectionId == collectionId, "incorrect collectionId specified");
+        // slither-disable-next-line timestamp
+        require(databonds[bondId].epochBondLastUpdated + epochLimitForUpdateBond <= epoch, "invalid databond update");
+
+        databonds[bondId].epochBondLastUpdated = epoch;
+        databonds[bondId].jobIds = jobIds;
+
+        collectionManager.updateCollection(collectionId, tolerance, aggregationMethod, power, jobIds);
+    }
+
+    function addJobsToCollection(
```

Comment:
calculate occurrence in places where databond collections are bing directly affected